fdbserver/Status.actor.cpp

```diff
@@ -2825,18 +2838,16 @@ ACTOR Future<StatusReply> clusterGetStatus(
 				statusObj["logs"] = tlogFetcher(&logFaultTolerance, db, address_workers);
 			}
 
-			if (configuration.present()) {
```

Comment:
Can you add an ASSERT_WE_THINK(configuration.present()) here, since you believe this configuration must exist?

Reply:
This whole code block is already within a

if (configuration.present())

See line 2786.

So I think it's trivial to not another "if (configuration.present())" within a "if (configuration.present())"

Reply:
The 65535th time I wish we could do `map` on the Flow `Optional`

Reply:
Oh wait, Flow does have that!

```
template <class R>
	Optional<R> map(std::function<R(T)> f) const {
		if (present()) {
			return Optional<R>(f(get()));
		} else {
			return Optional<R>();
		}
	}
```

https://github.com/apple/foundationdb/blob/master/flow/Arena.h#L229

Reply:
Sorry Xin, I didn't quite get your point. 

Another reason I removed this if statement was that on line 2826, it seems already assumed that configuration is present.

Reply:
Right. I meant since it has `map` implemented, one can then do something like `optionalValue.map([](T v){doStuff})` Just a nicer way to do things.